storage/gcp/gcp.go

```diff
@@ -85,25 +96,95 @@ func New(ctx context.Context, cfg Config) (*Storage, error) {
 	if err != nil {
 		return nil, fmt.Errorf("failed to create GCS storage: %v", err)
 	}
-
-	dbPool, err := spanner.NewClient(ctx, cfg.Spanner)
+	seq, err := newSpannerSequencer(ctx, cfg.Spanner)
 	if err != nil {
-		klog.Exitf("Failed to connect to Spanner: %v", err)
+		return nil, fmt.Errorf("failed to create Spanner sequencer: %v", err)
 	}
 
 	r := &Storage{
 		gcsClient: c,
 		projectID: cfg.ProjectID,
 		bucket:    cfg.Bucket,
-		dbPool:    dbPool,
 		objStore:  gcsStorage,
+		sequencer: seq,
 	}
+	// TODO(al): make queue options configurable:
+	r.queue = storage.NewQueue(time.Second, 256, r.sequencer.assignEntries)
 
-	if err := r.initDB(ctx); err != nil {
-		return nil, fmt.Errorf("failed to init DB: %v", err)
+	return r, nil
+}
+
+// tileSuffix returns either the empty string or a tiles API suffix based on the passed-in tile size.
+func tileSuffix(s uint64) string {
```

Comment:
Should this belong in layout? Can be moved later, in which case add a TODO.

Reply:
Yeah, this'll go away when Martin's API stuff is in.